core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -43,11 +43,12 @@ import scala.collection.mutable.ListBuffer
  *
  * A subclass of DelayedOperation needs to provide an implementation of both onComplete() and tryComplete().
  */
-abstract class DelayedOperation(override val delayMs: Long) extends TimerTask with Logging {
+abstract class DelayedOperation(override val delayMs: Long,
+    lockOpt: Option[Lock] = None) extends TimerTask with Logging {
```

Comment:
I suggest reconsidering all default arguments used for locks. Since it's crucial to get them right, it's generally safer to force callers to choose. My general recommendation is to avoid default arguments whenever correctness is involved.